app/controllers/service_controller.rb

```diff
@@ -69,6 +69,7 @@ def show
   def set_display
     @display = params[:display]
     @display ||= default_display unless pagination_or_gtl_request?
+    @display ||= 'generic_objects' if role_allows?(:feature => "generic_object_view") && @record.number_of(:generic_objects).positive?
```

Comment:
This looks strange. We want to set display to  'generic_objects' if the request is a `pagination_or_gtl_request`?

Reply:
`@display` was not set for All Generic Objects page - this is why it did not work.
"We want to set display to 'generic_objects' if the request is a pagination_or_gtl_request?"
... Hmm, I see there something slightly different: if the request is a `pagination_or_gtl_request`, `@display` still does not have to be set to 'generic_objects', because there is condition: `if role_allows?(:feature => "generic_object_view") && @record.number_of(:generic_objects).positive?` which has to be true for @display to be set to `generic_objects` ... Or did you mean to change this condition to something else? This worked for me the best. I also think that @display does not have to be always set, according to the original code.